tools/cache_perf_tool/sgwNodeAbstraction.go

```diff
@@ -54,6 +56,13 @@ func (s *sequenceAllocator) nextSeq() uint64 {
 		max := s.syncSeqMock.nextBatch(uint64(s.batchSize))
 		s.maxSeqInBatch = max
 		s.lastSeq = max - uint64(s.batchSize)
+		// channel has cap of 1 so to avoid blocking here we need select here
+		// the motivation to have cap 1 on this buffered channel is to sort of siluate dedupe from KV engine upon
+		// rapid updates to _sync:seq doc
+		select {
+		case s.syncSeqEvent <- struct{}{}:
```

Comment:
I'm not quite sure I understand what's happening here. There's nothing to drain `s.syncSeqEvent`, and the `default:` case is going to let things through even if the channel is full?

That seems like the same thing as this whole select not being here unless I did misunderstand something.

Reply:
Apologies this is soemeting that leaked out of the DCP side implementation, will remove form this PR